### DISCUSSION TITLE
MS2109 upgraded?

### DISCUSSION THREAD

**Discussion by DariBer** [awawa-dev/HyperHDR#276] [General]
I haven't seen this before, but could it be that the MS2109 is upgraded to USB 3.0? Saw this on Amazon, worth buy and see? https://www.amazon.co.uk/dp/B08L9KGNLT/ref=cm_sw_r_cp_api_i_5AAWVDW5TSTBZFE35KW5

@awawa-dev Is it possible to support your work with HyperHDR and the test you do and publish on the site, with money to buy 1 or 2 specific cards?

**Comment by awawa-dev** [comment]
This is popular fake USB3.0 version of MS2109. Why MS2109 is bound to USB2.0 is explained here with link to the manufacture diagram (embedded USB2.0 controller in MS2109 chip which probably can't be changed): [link](https://github.com/awawa-dev/HyperHDR/discussions/267#discussioncomment-2701954) and system logs should be similar to [link](https://github.com/awawa-dev/HyperHDR/discussions/267#discussioncomment-2701828)
Probably you found it already, but there are already our 2 big USB reviews out there you may find helpful.

[latency test](https://www.hyperhdr.eu/2021/10/usb-grabbers-grand-latency-test-part-i.html)
[HDR2SDR quality test](https://www.hyperhdr.eu/2022/04/usb-grabbers-hdr-to-sdr-quality-test.html)

If a new USB grabber appears on the market, I'm willing to test it (earnings from Github Sponsoring help a lot). In the past, Ezcap was a trendsetter but MS2109 turned everything upside down and their latest products are expensive and have some drawbacks to our use. Still, it's worth to watch them.

**Reply by DariBer** [reply]
If we put the money/price of the capture cards aside and just consider what you find is the best, is it still the MS2109, I find it little hard to take in what's best and what parts/function is better for the ambient light to work best. 

Like is the lack of USB 2 (bandwidth) a big issue, is it better to go with a escape 320 cause it has USB 3 and so on. 

So what's your personal favourite if you don't need to consider the price/cost of the cards

**Reply by awawa-dev** [reply]
You have got the results. Typical TV has a input lag around 100ms outside the game mode. If you disable smoothing with optimal configuration, you can still have LED strip reacting faster or simultaneously with the TV even using MS2109 thanks to HyperHDR optimization (and that was a major reason to fork this project from Hyperion NG). V18 brings another huge improvement for HDR tone mapping for MJPEG which was a major drawback for this format before. I consider the LED driver as even more important because for example: probably mostly chosen Wifi connection is terrible solution for our specific ambient lightning proijects. it's stable only in perfect environmental and every signal quality drop can ruin the experience or significantly increase the lag ...even if it only lasts a few seconds from time to time.

**Reply by DariBer** [reply]
ok, I'm building to use HyperSerialEsp8266 with D1 mini clone (CH340C). I have a MS2109 and Ezcap 320 (think it's called B revision) lying around. I have a FeinTech VSP01201 splitter. So I'm not sure what to use. 

What's your personal favourite at the moment?

**Reply by awawa-dev** [reply]
I've used Ezcap 320 for over a year: at first it was unusable due to bugs/flaws in the firmware, custom HDR LUT table is mandatory (the new one or the old from v16 release), the USB connectivity issue still remains (sometimes after re-plugin it connects as USB2.0 instead USB3.0, probably low quality USB-C socket in my 320, but how often you need to unplug the grabber from the Rpi?). I described it in my review on the blog, but overall was happy later with that grabber. So Ezcap 320 if you don't need DHCP and for simpler setup (less cables, no need for splitter) and the MS2109 + FeinTech otherwise. As you can find in the review Ezcap 320 has a bit weaker blue channel, so you must decide if it disturbs you. Currently I've switched to Ezcap 321 + Feintech, because I've already have them and new dedicated LUT table for Ezcap 321 is available. Earlier Ezcap 320 with its LUT tables provided better quality than Ezcap321/MS2109 using built-in default LUT tables. But there is a lot of cables and noticed Feintech sometimes (like one in 40-60 cases) has a problem with switching resolution when the movie starts to play. For me it's a minor issue. If the build quality of the Ezcap 331 wasn't that low (risk of overheating) and the need of USB3.0 (may not be detected on USB2.0), I would recommend it over the Ezcap 321.

**Reply by DariBer** [reply]
Sorry, I miixed them up, 320 and 321, I have a Ezcap 321A.
